Utils/HandlerUtil.py

```diff
@@ -61,26 +61,62 @@
 import base64
 import json
 import time
+
+from xml.etree import ElementTree
 from os.path import join
 from Utils.WAAgentUtil import waagent
 from waagent import LoggerInit
 
 DateTimeFormat = "%Y-%m-%dT%H:%M:%SZ"
 
+MANIFEST_XML = "manifest.xml"
+
 class HandlerContext:
     def __init__(self,name):
         self._name = name
         self._version = '0.0'
         return
 
 class HandlerUtility:
-    def __init__(self, log, error, short_name):
+    def __init__(self, log, error, s_name=None, l_name=None, version=None):
         self._log = log
         self._error = error
-        self._short_name = short_name
 
+        if s_name is None or l_name is None or version is None:
+            (long_name, short_name, ext_version) = self._get_extension_info()
+            self._short_name = short_name
+            self._extension_version = ext_version
+            self._log_prefix = '[%s-%s] ' % (long_name, ext_version)
+        else:
+            self._short_name = s_name
+            self._extension_version = version
+            self._log_prefix = '[%s-%s] ' % (l_name, version)
+
+    def get_extension_version(self):
+        return self._extension_version
+    
     def _get_log_prefix(self):
-        return '[%s-%s]' %(self._context._name, self._context._version)
+        return self._log_prefix if self._log_prefix else '[%s-%s] ' % (self._context._name, self._context._version)
```

Comment:
That's true now, but wasn't earlier. If the new code change guarantees self._log_prefix to be non-None all the time, we can change this code accordingly as well. I still don't understand "skipping the constructor and having self._log_prefix == None".

Again, the motivation for this original workaround was to retain the existing behavior as much as possible. I'm not saying the existing behavior is correct. The thing with the original implementation is that the self._context can change anytime, resulting in the varying extension name/ver in logs. And I didn't want to compose the string everytime, when it's fixed with the explicitly passed extension name and version at the construction time.